sinks/sinks.go

```diff
@@ -91,4 +93,6 @@ type SinkRepository interface {
 	RetrieveById(ctx context.Context, key string) (Sink, error)
 	// Remove a existing Sink by id
 	Remove(ctx context.Context, owner string, key string) error
+	// RetrieveToValidate validates a sink
+	RetrieveToValidate(ctx context.Context, name, owner string) error
```

Comment:
why is this needed?

Reply:
We also check if the name already exists on database. So, before we save it, we can check if the name is available, and not duplicated.